File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/movie/model/MovieVideos.java
package dev.reuss.tmdb.domain.movie.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.common.video.VideosResponse;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Videos for a TMDB movie.
 *
 * @param id      TMDB movie id
 * @param results movie videos
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record MovieVideos(
        int id,
        List<MovieVideo> results
) implements VideosResponse<MovieVideo>, TmdbModel {

    public MovieVideos {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/movie/model/MovieVideos.kt
package dev.reuss.tmdb.domain.movie.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.common.video.VideosResponse

/**
 * Videos for a TMDB movie.
 *
 * @property id TMDB movie id
 * @property results movie videos
 */
@JvmRecord
data class MovieVideos(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("results")
    override val results: List<MovieVideo> = emptyList(),
) : VideosResponse<MovieVideo>,
    TmdbModel
